scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,525 @@
+
+    return dbOptions;
+};
+
+/*
+ * Loads the static data from disk, and creates a separate reference to the
+ * views document.
+ * @param {Object} options - Object that has the path to the directory
+ *                           containing the static data. On output, contains
+ *                           members for the new views and other static data.
+ */
+gpii.dataLoader.loadStaticDataFromDisk = function (options) {
+    var data = gpii.dataLoader.getDataFromDirectory(options.staticDataDir);
+    var views = fluid.find(data, function (anElement) {
+        if (anElement._id && anElement._id === "_design/views") {
+            return anElement;
+        } else {
+            return undefined;
+        }
+    });
+    options.staticData = data;
+    options.newViews = views;
+    fluid.log("Retrieved static data from: '" + options.staticDataDir + "'");
+    fluid.log("\tViews data " + ( views ? "retrieved." : "missing." ));
+};
+
+/*
+ * Utility to configure a step:  creates a response callback, binds it to an
+ * http database request, and configures a promise to resolve/reject when the
+ * response callback finishes or fails.
+ * @param {Object} details - Specific information for the request and response,
+ *                           such as error messages, urls, POST data, and so on.
+ *                           These details are set appropriately by the caller.
+ * @param {Object} options - Additional options.
+ * @return {Promise} - A promise that resolves the configured step.
+ */
+gpii.dataLoader.configureStep = function (details, options) {
+    var togo = fluid.promise();
+    var response = gpii.dataLoader.createResponseHandler(
+        details.responseDataHandler,
+        options,
+        togo,
+        details.responseErrMsg
+    );
+    var request;
+    if (details.dataToPost) {
+        request = gpii.dataLoader.createPostRequest(
+            details.dataToPost, response, options
+        );
+    } else {
+        request = gpii.dataLoader.queryDatabase(
+            details.requestUrl, response, details.requestErrMsg
+        );
+    }
+    request.end();
+    return togo;
+};
+
+/*
+ * Create the step that loads the static data into the database.
+ * @param {Object} options - Object that has the static data to load.
+ * @return {Promise} - A promise that resolves loading the static data.
+ */
+gpii.dataLoader.createStaticDataStep = function (options) {
+    var details = {
+        dataToPost: options.staticData,
+        responseDataHandler: function (responseString, options) {
+            fluid.log("Loading static data from '" + options.staticDataDir + "'");
+            return "Uploaded static data.";
+        },
+        responseErrMsg: "Error loading static data into database: "
+    };
+    return gpii.dataLoader.configureStep(details, options);
+};
+
+/*
+ * Create the step that retrieves the current views from the database.
+ * @param {Object} options - Object containing the views URL into the database.
```

Comment:
Be a little clearer about what fields are expected/allowed in this options block, possibly using a JSDocs @typedef

Reply:
See below.